hamilton/lifecycle/default.py

```diff
@@ -584,10 +598,43 @@ def run_to_execute_node(
         # and truncate it/provide sentinels for every failure)
         # TODO -- decide what to do with collect
         """Executes a node. If the node fails, returns the sentinel value."""
-        for key, value in node_kwargs.items():
-            if value == self.sentinel_value:  # == versus is
-                return self.sentinel_value  # cascade it through
+        default_return = [self.sentinel_value] if is_expand else self.sentinel_value
```

Comment:
Does empty list not work? Or do you think a single list is cleaner?

Reply:
I just learned that Hamilton behaves nicely when an expand yields an empty list and takes that empty list to the collect! 

Since sentinel injection is a feature, a user might want to use it after a collect node. If the collect node got an empty list, there wouldn't be a way to tell if it's because the expand node didn't yield anything or if the expand node was skipped. So I lean toward single list being more explicit about what happened.

Reply:
Heh I had to test to ensure, and yes it turns out it does :)

I think that does make sense -- there's something very clean about a single sentinel value. That said, it *will* still execute on the executor (if this is, say, ray, that would be a slightly more expensive operation). But it'll look better in the UI and cascade it through. So yeah, I think a single list is OK here.